packages/jazz-tools/src/coValues/coList.ts

```diff
@@ -446,6 +449,77 @@ export class CoList<Item = any> extends Array<Item> implements CoValue {
     ): InstanceType<Cl> {
         return cl.fromRaw(this._raw) as InstanceType<Cl>;
     }
+
+    asPlainData<L extends CoList>(this: L): RecursiveCoMapInit<L>;
+    asPlainData<L extends CoList, Depth>(
+        this: L,
+        depth: Depth & DepthsIn<L>,
+    ): Promise<RecursiveCoMapInit<L> | undefined>;
+    asPlainData<L extends CoList, Depth>(
+        this: L,
+        depthOrSeen?:
+            | (Depth & DepthsIn<L>)
+            | WeakMap<CoMap | CoList, RecursiveCoMapInit<L>>,
+    ): RecursiveCoMapInit<L> | Promise<RecursiveCoMapInit<L> | undefined> {
+        if (depthOrSeen instanceof WeakMap) {
+            return this.asPlainDataSync(depthOrSeen);
+        } else if (depthOrSeen === undefined) {
+            return this.asPlainDataSync(new WeakMap());
+        } else {
+            return this.asPlainDataAsync(depthOrSeen, new WeakMap());
+        }
+    }
+
+    private asPlainDataSync<L extends CoList>(
+        this: L,
+        seen: WeakMap<CoMap | CoList, RecursiveCoMapInit<UnCo<L[number]>>[]>,
+    ): RecursiveCoMapInit<L> {
+        if (seen.has(this)) {
+            return seen.get(this) as RecursiveCoMapInit<L>;
+        }
```

Comment:
**suggestion:** we can avoid to use the type cast this way:

```
const circularRef = seen.get(this);

if (circularRef) return circularRef;
```

Also since this codepath can become very hot when serializing big chunks of state using this approach we reduce the number of access to the map